tests/lean/run/coindBisim.lean

```diff
@@ -0,0 +1,87 @@
+structure FSM where
+  S : Type
+  d : S → Nat → S
+  A : S → Prop
+
+-- Example of a coinductive predicate defined over FSMs
+
+coinductive Bisim (fsm : FSM) : fsm.S → fsm.S → Prop :=
+  | step {s t : fsm.S} :
+    (fsm.A s ↔ fsm.A t)
+    → (∀ c, Bisim fsm (fsm.d s c) (fsm.d t c))
+    → Bisim fsm s t
+
+/--
+info: inductive Bisim.Shape : (fsm : FSM) → ((fsm : FSM) → fsm.S → fsm.S → Prop) → fsm.S → fsm.S → Prop
+number of parameters: 4
+constructors:
+Bisim.Shape.step : ∀ {fsm : FSM} {Bisim : (fsm : FSM) → fsm.S → fsm.S → Prop} {s t : fsm.S},
+  (fsm.A s ↔ fsm.A t) → (∀ (c : Nat), Bisim fsm (fsm.d s c) (fsm.d t c)) → Bisim.Shape fsm Bisim s t
+-/
```

Comment:
This doesn't feel right. `FSM` is a fixed parameter to `Bisim` (it is given as a binder, before the `:`), and should *not* show up in the motive. That is, I would've expected this to have type
`(fsm : FSM) → (fsm.S → fsm.S → Prop) → fsm.S → fsm.S → Prop`